src/timer/timer_game_economy.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file timer_game_economy.cpp
+ * This file implements the timer logic for the game-economy-timer.
+ */
+
+#include "stdafx.h"
+#include "openttd.h"
+#include "timer.h"
+#include "timer_game_economy.h"
+#include "timer_game_calendar.h"
+#include "vehicle_base.h"
+#include "linkgraph/linkgraph.h"
+
+#include "safeguards.h"
+
+TimerGameEconomy::Year TimerGameEconomy::year = {};
+TimerGameEconomy::Month TimerGameEconomy::month = {};
+TimerGameEconomy::Date TimerGameEconomy::date = {};
+TimerGameEconomy::DateFract TimerGameEconomy::date_fract = {};
+
+/**
+ * Converts a Date to a Year, Month & Day.
+ * @param date the date to convert from
+ * @param ymd  the year, month and day to write to
+ */
+/* static */ void TimerGameEconomy::ConvertDateToYMD(TimerGameEconomy::Date date, YearMonthDay *ymd)
+{
+	TimerGameCalendar::YearMonthDay calendar_ymd;
+	calendar_ymd.day = ymd->day;
+	calendar_ymd.month = ymd->month;
+	calendar_ymd.year = ymd->year;
+	TimerGameCalendar::ConvertDateToYMD(date, &calendar_ymd);
+}
```

Comment:
This does not look like it is doing the right thing.
*ymd is not actually written to here.

Starting a new game with this PR seems to result in TimerGameEconomy::year/month having silly values.